src/connectdlg.cpp

```diff
@@ -529,15 +538,39 @@ void CConnectDlg::OnServerAddrEditTextChanged ( const QString& )
 
 void CConnectDlg::OnCustomCentralServerAddrChanged()
 {
+
+    QString strPreviousSelection = cbxDirectoryServer->currentText();
     UpdateDirectoryServerComboBox();
-    // only update list if a custom server list is selected
+    // after updating the combobox, we must re-select the previous directory selection
+
     if ( pSettings->eCentralServerAddressType == AT_CUSTOM )
     {
-        // TODO: detect if the currently select custom directory still exists in the now potentially re-ordered vector,
-        //       if so, then change to its new index.  Issue #1899
-        pSettings->eCentralServerAddressType = static_cast<ECSAddType> ( AT_DEFAULT );
-        pSettings->iCustomDirectoryIndex     = 0;
-        RequestServerList();
+        // check if the currently select custom directory still exists in the now potentially re-ordered vector,
+        // if so, then change to its new index.  (addresses Issue #1899)
+        int iNewIndex = cbxDirectoryServer->findText ( strPreviousSelection );
```

Comment:
Is it possible to have an entry with the same ip/domain twice? Probably not. 

According to the Qt docs:

> Returns the index of the item containing the given text; otherwise returns -1.

What happens if we have

wold.co.uk and wold.co as directories?

Reply:
There are options on findText do do a partial match.  But the default (if I understand the docs correctly) is QT::MatchExactly

Reply:
> Is it possible to have an entry with the same ip/domain twice?

Servers are unique by IP:port.  I run up to a dozen (although most of them are directory servers themselves).